### PR TITLE
Migrating the helm deployments to Go templates

### PR DESCRIPTION
**Pull request by Vidya2606**
# Description

Migrate Deployment templates to go templates

Fixes # (issue)
Feature # (details)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Is it a  breaking change which will impact consuming tool(s).

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pkg/osutil/osutil.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package osutil
 import (
 	"bytes"
 	"fmt"
+	log "github.com/sirupsen/logrus"
```
Comment (bosesuneha) [inline pkg/osutil/osutil.go:6]: nit: import order

File: template/deployments/helm/charts/templates/_helpers.tpl [hunk 2]
```diff
@@ -1,17 +1,16 @@
 {{/*
 Expand the name of the chart.
 */}}
-{{- define "{{APPNAME}}.name" -}}
-{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" }}
-{{- end }}
+{{ printf "{{- define \"%s.name\" -}}" .APPNAME }}
+{{ printf "{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix \"-\" }}\n{{- end }}" }}
```
Comment (bfoley13) [inline template/deployments/helm/charts/templates/_helpers.tpl:5]: Lets keep it consistent, if you dont have a draft variable replacement happening then lets just use the comment wrapping method

File: template/deployments/helm/charts/templates/service.yaml [hunk 3]
```diff
@@ -1,19 +1,13 @@
...
-  namespace: {{ .Values.namespace }}
+    {{print "{{ toYaml .Values.service.annotations | nindent 4 }}\nnamespace: {{ .Values.namespace }}" }}
 spec:
-  type: {{ .Values.service.type }}
-  ports:
-    - port: {{ .Values.service.port }}
-      targetPort: {{ .Values.containerPort }}
-      protocol: TCP
-      name: svchttp
+  {{ printf "spec:\n  type: {{ .Values.service.type }}\n  ports:\n    - port: {{ .Values.service.port }}\n      targetPort: {{ .Values.containerPort }}\n      protocol: TCP\n      name: svchttp" }}
```
Comment (bfoley13) [inline template/deployments/helm/charts/templates/service.yaml:11]: I think you could use the same {{` `}}  wrappes in this case as well

File: template/deployments/helm/charts/templates/deployment.yaml [hunk 4]
```diff
@@ -1,28 +1,28 @@
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: {{ include "{{APPNAME}}.fullname" . }}
+  name: {{ printf "{{ include \"%s.fullname\" . }}" .APPNAME }}
   labels:
-    {{- include "{{APPNAME}}.labels" . | nindent 4 }}
-    kubernetes.azure.com/generator: {{GENERATORLABEL}}
+    {{ printf "{{- include \"%s.labels\" . | nindent 4 }}" .APPNAME }} {{`
```
Comment (bfoley13) [inline template/deployments/helm/charts/templates/deployment.yaml:6]: Its beneficial to have the {{` and `}} be on new lines.  You can probably use something like {{-` and `-}} to remove the whitespace created by them in the result

Comment (bfoley13) [reply template/deployments/helm/charts/templates/deployment.yaml:6]: Ahh markdown is making it odd to understand but I mean this only for the comment components and not the ones containing `printf`

Comment (Vidya2606) [reply template/deployments/helm/charts/templates/deployment.yaml:6]: yeah I understood! I will update it

**Review by bosesuneha — APPROVED** [review]
lgtm

**Review by Tatsinnit — APPROVED** [review]
Looks good to my bare eyes and without knowing the context, please get owners to take a Quick Look. Thanks heaps for the ping. ❤️🙏☕️

**Review by davidgamero — APPROVED** [review]
lgtm

**Pull request merged**